dev/plugins/hu.elte.txtuml.layout.visualizer/src/hu/elte/txtuml/layout/visualizer/algorithms/links/GraphSearch.java

```diff
@@ -313,6 +313,79 @@ else if (dir.equals(Direction.opposite(sub)))
 
 		return result;
 	}
+	
+	public boolean isReflexivePath(ArrayList<Node> path)
+	{
+		return path.size() == 4 
+				&& 
+				((path.get(0).getTo().getX() == path.get(3).getFrom().getX() 
+				&& Math.abs(path.get(0).getTo().getY() - path.get(3).getFrom().getY()) == 2) 
+				|| 
+				(path.get(0).getTo().getY() == path.get(3).getFrom().getY() 
+				&& Math.abs(path.get(0).getTo().getX() - path.get(3).getFrom().getX()) == 2));					
+	}
+	
+	public ArrayList<Node> extendReflexivePath(ArrayList<Node> result)
```

Comment:
Do we need to get and return ArrayList here? I think List<Node> is better and more general.